code/game/objects/items/weapons/RCD.dm

```diff
@@ -282,7 +288,7 @@ RCD
 
 
 /obj/item/weapon/rcd/afterattack(atom/A, mob/user, proximity)
-	if(!proximity)
+	if(!proximity && !ranged)
```

Comment:
I kind of get it:
1) I dont get why this is more efficient than a simple if-statement

2) Im not sure how the ARCD would "override" the proc